### PR TITLE
update 0-stor client creation to use new 0-stor policy

### PR DESCRIPTION
**Pull request by zaibon**

### CODE REVIEW

File: Gopkg.lock [hunk 1]
```diff
@@ -478,6 +478,6 @@
 [solve-meta]
   analyzer-name = "dep"
   analyzer-version = 1
-  inputs-digest = "b41c64c1ea6cc97c3a2d89a77a294153ecd41ed481b1faf682bef926b57b2c74"
+  inputs-digest = "8013d1253210767c6e604d4e39d9a5adf9baac73dbe7ee1faad408dcedf9eb70"
```
Comment (GlenDC) [inline Gopkg.lock]: This should have been part of the vendor commit though, next time better luck 💃

File: tlog/stor/client.go [hunk 2]
```diff
@@ -161,45 +157,21 @@ func (c *Client) LastHash() []byte {
...
-		},
+func newStorClientConf(conf Config) storclient.Policy {
+	return storclient.Policy{
+		Organization:           conf.Organization,
+		Namespace:              conf.Namespace,
+		DataShards:             conf.ZeroStorShards,
+		MetaShards:             conf.MetaShards,
+		IYOAppID:               conf.IyoClientID,
+		IYOSecret:              conf.IyoSecret,
+		Protocol:               "grpc",
```
Comment (GlenDC) [inline tlog/stor/client.go]: can we turn this into constants? Doesn't have to be in this PR though, just would be nice if you can give that a thought and if you agree make an issue for it in the 0-stor repo. Perhaps you have good reason not to turn it into a constant though.

File: tlog/stor/client.go [hunk 3]
```diff
@@ -161,45 +157,21 @@ func (c *Client) LastHash() []byte {
...
+		Compress:               true,
```
Comment (GlenDC) [inline tlog/stor/client.go]: Is it still possible to choose the compression and encryption algorithm or is it now hardcoded for all users?

File: tlog/stor/client.go [hunk 4]
```diff
@@ -161,45 +157,21 @@ func (c *Client) LastHash() []byte {
...
+		Encrypt:                true,
+		EncryptKey:             conf.EncryptPrivKey,
+		ReplicationNr:          0, //force to use distribution
```
Comment (GlenDC) [inline tlog/stor/client.go]: Can you link me to docs or explain what is the difference between Replication vs distribution in this context?

File: Gopkg.lock (hunk 1 above)
Comment (zaibon) [reply Gopkg.lock]: yep, still learning the dep tool.

File: tlog/stor/client.go (hunk 4 above)
Comment (zaibon) [reply tlog/stor/client.go]: So more detail about how policy works
https://godoc.org/github.com/zero-os/0-stor/client#Policy

File: tlog/stor/client.go (hunk 3 above)
Comment (zaibon) [reply tlog/stor/client.go]: It's not configurable anymore at the moment.

File: tlog/stor/client.go (hunk 2 above)
Comment (zaibon) [reply tlog/stor/client.go]: Should be a constant, you are right.

File: tlog/stor/client.go (hunk 4 above)
Comment (GlenDC) [reply tlog/stor/client.go]: From those docs I wouldn't have guessed I have to set 0 to both properties to use distribution though.

```go
    // Number of replication to create when writting
    ReplicationNr int `yaml:"replication_nr"`
    // if data size is smaller than ReplicationMaxSize then data
    // will be replicated ReplicationNr time
    // if data is bigger, distribution will be used if configured
    ReplicationMaxSize int `yaml:"replication_max_size"`
```

Also I still don't understand the concepts of distribution vs replication in this context. I mean I have an idea, but these things can be ambiguous so I rather I have it clear in my head.

**Comment by GlenDC** [comment]
@zaibon is it important that this still goes in alpha-8, instead of going straight to master?

**Comment by zaibon** [comment]
Would be nice if we can integrate that in alpha8 yes.

**Comment by GlenDC** [comment]
> Would be nice if we can integrate that in alpha8 yes.

I can do it, it's just that I had hoped we would stop pushing any unplanned changes to alpha-8. But if you want, I'll get this one through.

File: tlog/stor/client.go (hunk 4 above)
Comment (zaibon) [reply tlog/stor/client.go]: You don't need to put it to 0. You can have them both enabled. Then based on the size of the data you send, or replication or distribution will be used.

Replication is writing same data multiple time
Distribution is erasure coding,

Comment (GlenDC) [reply tlog/stor/client.go]: Yeah I know, by default the int value in `Go` is `0`, so I suppose you're just being explicit here.

> Replication is writing same data multiple time
> Distribution is erasure coding

Thanks that clarifies it for me. As long as you only ever use Distribution and Erasure encoding, it should be simple enough to keep understanding it I suppose.

Comment (zaibon) [reply tlog/stor/client.go]: >Yeah I know, by default the int value in Go is 0, so I suppose you're just being explicit here.

indeed, this is just to be explicit

Comment (zaibon) [reply tlog/stor/client.go]: I'll improve the documentation regarding the policy of 0-stor and how thing works together

Comment (GlenDC) [reply tlog/stor/client.go]: Yeah sure, even just linking to the relevant `/docs` documentation might already be good enough.

**Pull request merged**